### PR TITLE
Remove authType config

### PR DESCRIPTION
**Pull request by nwalters512**
As discussed on #8198.

This PR aims to maintain existing devMode behavior as much as possible. If someone opens PrairieLearn for the first time in their browser, we'll automatically authenticate them as `config.authUid` and friends. The only change is when the user clicks "Log out". In this case, we clear the `pl_authn` cookie and set a special `pl_disable_auto_authn` cookie. This prevents the automatic authentication behavior and allows the user to experience the site as a logged-out user. If they visit the login page, they can use the "bypass" auth option to set the cookie to what's configured in `config.authUid` (see `pages/authLoginDev`).

This will all become much more useful in #8198, where the login page will allow authenticating as arbitrary users in dev mode.

This is core auth code, so a thorough and careful review would be appreciated 🙏

### CODE REVIEW

File: apps/prairielearn/src/middlewares/authn.js [hunk 1]
```diff
@@ -110,7 +118,10 @@ module.exports = asyncHandler(async (req, res, next) => {
   }
   if (authnData == null) {
     // We failed to authenticate.
-    //
+
+    // Clear the pl_authn cookie in case it was bad
+    res.clearCookie('pl_authn');
```
Comment (nwalters512) [inline apps/prairielearn/src/middlewares/authn.js:123]: This change wasn't necessary; it just seemed like a good idea to unconditionally clear that authentication cookie if we failed to decode it.

File: apps/prairielearn/src/middlewares/authzCourseOrInstance.js [hunk 2]
```diff
@@ -33,10 +33,7 @@ module.exports = function (req, res, next) {
           allow_example_course_override: true,
           ip: req.ip,
           req_date: res.locals.req_date,
-          req_mode:
-            config.authType === 'none' && req.cookies.pl_test_mode
-              ? req.cookies.pl_test_mode
-              : null,
+          req_mode: config.devMode && req.cookies.pl_test_mode ? req.cookies.pl_test_mode : null,
```
Comment (nwalters512) [inline apps/prairielearn/src/middlewares/authzCourseOrInstance.js:36]: This check should be equivalent for all intents and purposes.

File: apps/prairielearn/src/middlewares/clearCookies.js [hunk 3]
```diff
@@ -1,6 +1,11 @@
 const _ = require('lodash');
 
-const cookies_to_ignore = ['pl_authn', 'pl_assessmentpw', 'pl_access_as_administrator'];
+const cookies_to_ignore = [
+  'pl_authn',
+  'pl_assessmentpw',
+  'pl_access_as_administrator',
+  'pl_disable_auto_authn',
```
Comment (nwalters512) [inline apps/prairielearn/src/middlewares/clearCookies.js:7]: I spent waaaaay too long wondering why my cookies were getting cleared immediately after being set 😭

**Review by mwest1066 — APPROVED** [review]

File: apps/prairielearn/src/middlewares/authn.js [hunk 4]
```diff
@@ -64,16 +64,24 @@ module.exports = asyncHandler(async (req, res, next) => {
...
-  if (config.authType === 'none') {
+  // In dev mode, by default, we'll authenticate the user automatically using
+  // the UID, name, and UIN specified in the config. This is to reduce the
+  // friction of getting started with PrairieLearn.
+  //
+  // If an authentication cookie is already present, we won't set a new one.
+  //
+  // If the user clicks "Log out" in dev mode, we'll set a special cookie to
+  // prevent this automatic authentication. The user will get bounced to the
+  // login page like they would in production. Then can then use the "bypass"
```
Comment (jonatanschroeder) [inline apps/prairielearn/src/middlewares/authn.js:75]:
```suggestion
  // login page like they would in production. They can then use the "bypass"
```

**Review by jonatanschroeder — APPROVED** [review]

**Pull request merged**